File: backend/src/main/java/com/tripfriend/domain/member/member/dto/MemberResponseDto.java
package com.tripfriend.domain.member.member.dto;

import com.tripfriend.domain.member.member.entity.AgeRange;
import com.tripfriend.domain.member.member.entity.Gender;
import com.tripfriend.domain.member.member.entity.Member;
import com.tripfriend.domain.member.member.entity.TravelStyle;
import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
public class MemberResponseDto {

    private Long id;
    private String username;
    private String email;
    private String nickname;
    private String profileImage;
    private Gender gender;
    private AgeRange ageRange;
    private TravelStyle travelStyle;
    private String aboutMe;
    private Double rating;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String authority;

    public static MemberResponseDto fromEntity(Member member) {

        return new MemberResponseDto(
                member.getId(),
                member.getUsername(),
                member.getEmail(),
                member.getNickname(),
                member.getProfileImage(),
                member.getGender(),
                member.getAgeRange(),
                member.getTravelStyle(),
                member.getAboutMe(),
                member.getRating(),
                member.getCreatedAt(),
                member.getUpdatedAt(),
                member.getAuthority()
        );
    }
}

File: backend/src/main/java/com/tripfriend/domain/member/member/dto/MemberResponseDto.kt
package com.tripfriend.domain.member.member.dto

import com.tripfriend.domain.member.member.entity.AgeRange
import com.tripfriend.domain.member.member.entity.Gender
import com.tripfriend.domain.member.member.entity.Member
import com.tripfriend.domain.member.member.entity.TravelStyle
import java.time.LocalDateTime

data class MemberResponseDto(
    val id: Long?,
    val username: String,
    val email: String,
    val nickname: String,
    val profileImage: String?,
    val gender: Gender,
    val ageRange: AgeRange,
    val travelStyle: TravelStyle,
    val aboutMe: String?,
    val rating: Double,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val authority: String
) {
    companion object {
        @JvmStatic
        fun fromEntity(member: Member): MemberResponseDto {
            return MemberResponseDto(
                id = member.id,
                username = member.username,
                email = member.email,
                nickname = member.nickname,
                profileImage = member.profileImage,
                gender = member.gender,
                ageRange = member.ageRange,
                travelStyle = member.travelStyle,
                aboutMe = member.aboutMe,
                rating = member.rating,
                createdAt = member.createdAt,
                updatedAt = member.updatedAt,
                authority = member.authority
            )
        }
    }
}
